iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -261,66 +274,78 @@ const T& vector<T, Capacity>::front() const noexcept
-T& vector<T, Capacity>::back() noexcept
+inline typename vector<T, Capacity>::reference vector<T, Capacity>::back() noexcept
 {
     if (empty())
     {
         std::cerr << "Attempting to access the back of an empty vector!" << std::endl;
         std::terminate();
     }
-    return at(size() - 1u);
+    return at(this->size() - 1u);
 }
 
 template <typename T, uint64_t Capacity>
-const T& vector<T, Capacity>::back() const noexcept
+inline typename vector<T, Capacity>::const_reference vector<T, Capacity>::back() const noexcept
 {
     if (empty())
     {
         std::cerr << "Attempting to access the back of an empty vector!" << std::endl;
         std::terminate();
     }
-    return at(size() - 1);
+    return at(this->size() - 1);
 }
 
 template <typename T, uint64_t Capacity>
-inline typename vector<T, Capacity>::iterator vector<T, Capacity>::begin()
+inline typename vector<T, Capacity>::iterator vector<T, Capacity>::begin() noexcept
 {
-    return reinterpret_cast<iterator>(&(m_data[0]));
+    return reinterpret_cast<iterator>(this->data());
 }
 
 template <typename T, uint64_t Capacity>
-inline typename vector<T, Capacity>::const_iterator vector<T, Capacity>::begin() const
+inline typename vector<T, Capacity>::const_iterator vector<T, Capacity>::begin() const noexcept
 {
-    return reinterpret_cast<const_iterator>(&(m_data[0]));
+    return reinterpret_cast<const_iterator>(this->data());
 }
 
 template <typename T, uint64_t Capacity>
-inline typename vector<T, Capacity>::iterator vector<T, Capacity>::end()
+inline typename vector<T, Capacity>::iterator vector<T, Capacity>::end() noexcept
 {
-    return reinterpret_cast<iterator>((&(m_data[0]) + m_size)[0]);
+    return this->data() + this->size();
 }
 
 template <typename T, uint64_t Capacity>
-inline typename vector<T, Capacity>::const_iterator vector<T, Capacity>::end() const
+inline typename vector<T, Capacity>::const_iterator vector<T, Capacity>::end() const noexcept
 {
-    return reinterpret_cast<const_iterator>((&(m_data[0]) + m_size)[0]);
+    return reinterpret_cast<const_iterator>(this->data() + this->size());
 }
 
 template <typename T, uint64_t Capacity>
-inline typename vector<T, Capacity>::iterator vector<T, Capacity>::erase(iterator position)
+inline typename vector<T, Capacity>::iterator vector<T, Capacity>::erase(const_iterator position)
 {
-    if (begin() <= position && position < end())
+    /// If position < begin(), this shall return end(), which is guaranteed:
+    /// WLOG, let vector v have Capacity=(2^64-begin())/sizeof(T)-sizeof(m_size)
+    /// (note that iterators, index, and Capacity are all limited by 64 bit address space)
+    /// With position=0, position-begin()==(2^64-begin())/sizeof(T) > Capacity => returns end()
+    uint64_t index = static_cast<uint64_t>(position - begin());
+    return erase(index);
+}
+
+template <typename T, uint64_t Capacity>
+inline typename vector<T, Capacity>::iterator vector<T, Capacity>::erase(uint64_t index)
```

Comment:
Please add const for the parameter here and in the header.

Reply:
Ok.